lib/pharos/phases/configure_kubelet.rb

```diff
@@ -24,6 +24,7 @@ def initialize(host, config)
 
       def call
         configure_cni
+        configure_kubelet_proxy if @host.role == 'worker'
         configure_kube
```

Comment:
I don't understand how this would work... the `configure-kubelet-proxy.sh` starts the kubelet with `/etc/systemd/system/kubelet.service.d/5-pharos-kubelet-proxy.conf` in place, but then this immediately runs `configure-kube.sh` which nukes that file and restarts the kubelet?

I guess the kubelet-proxy pod must remain running even when the kubelet gets restarted with the kubeadm configuration... feels a bit sketchy, but I guess it works?

Reply:
What happens if the node setup fails after the `ConfigureKubelet`, but before the `JoinNode` completes... if the host gets rebooted, then the kubelet-proxy pod is no longer going to be running, but the `configure-kubelet-proxy.sh` is not going to re-configure the kubelet to start it either?

Not the most robust way to run the proxy, I think... should it maybe be something like a systemd service with `rkt` instead, so it's independent of the kubelet?

Reply:
I guess that would just mean that the `Waiting kubelet-proxy to launch on port 6443...` bit is perhaps unnecessary, and it could actually  cause`pharos-cluster up` to break when retrying after errors, whereas just blindly running `kubeadm join` would fix it and allow the kubelet-proxy to start up again?

Reply:
Port wait is there to avoid timing issues. But it might actually work by blindly running `kubeadm join`. Need to test this after I finish merging master to this branch.

Reply:
Blindly running `kubeadm join` does not work.